File: services/api/app/schemas/account.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class NeteaseSessionCookie(BaseModel):
    name: str = Field(min_length=1)
    value: str = Field(min_length=1)
    domain: str | None = None
    path: str | None = None


class NeteaseSessionSnapshot(BaseModel):
    cookies: list[NeteaseSessionCookie]
    captured_at: datetime | None = None


class NeteaseAccountProfile(BaseModel):
    user_id: str
    nickname: str
    avatar_url: str | None = None


class NeteaseAccountSessionResponse(BaseModel):
    status: Literal["logged_in", "logged_out", "expired"]
    profile: NeteaseAccountProfile | None = None
    checked_at: datetime

File: shared/contracts/src/account.ts
export type NeteaseAccountSessionStatus =
  | "logged_in"
  | "logged_out"
  | "expired";

export interface NeteaseAccountProfile {
  user_id: string;
  nickname: string;
  avatar_url?: string;
}

export interface NeteaseSessionCookie {
  name: string;
  value: string;
  domain?: string;
  path?: string;
}

export interface NeteaseSessionSnapshot {
  cookies: NeteaseSessionCookie[];
  captured_at?: string;
}

export interface NeteaseAccountSessionResponse {
  status: NeteaseAccountSessionStatus;
  profile?: NeteaseAccountProfile;
  checked_at: string;
}
